### ISSUE TITLE
Error recognizing modules behind a macro

### ISSUE THREAD

**Issue by krojew** [intellij-rust/intellij-rust#4714]
## Environment

* **IntelliJ Rust plugin version:** 0.2.111.2147-193
* **Rust toolchain version:** 1.40 nightly
* **IDE name and version:** CLion 2019.3
* **Operating system:** Windows 10

## Problem description

One of the most popular crates, tokio, has modules which cannot be parsed by the plugin. Examples include the tokio::sync::mpsc module, which is imported inside a macro:

```rust
cfg_sync! {
    mod barrier;
    pub use barrier::{Barrier, BarrierWaitResult};

    pub mod mpsc;

    mod mutex;
    pub use mutex::{Mutex, MutexGuard};

    pub mod oneshot;

    pub(crate) mod semaphore;

    mod task;
    pub(crate) use task::AtomicWaker;

    pub mod watch;
}
```

mpsc is not recognized by the plugins at all.

## Steps to reproduce

Use tokio, try to include tokio::sync::mpsc::Whatever, see error.

**Comment by tdyas** [comment]
I am also seeing this issue with the tokio crate. My project is at https://github.com/tdyas/tftp-rs. Load it into IntelliJ and you will see that references to `UdpSocket` (`tokio::net::UdpSocket`) are red because the Rust plugin cannot find a definition.

**Comment by stjepangolemac** [comment]
Yeah it can't detect the tokio TcpListener too, just keeps importing the std::net one.

**Comment by lights0123** [comment]
This issue is also demonstrated in the [uom](https://crates.io/crates/uom) crate, which has no (useful) assistance by this plugin as all (useful) modules are imported by macros.

**Comment by SpyMachine** [comment]
reqwest 0.10 now has this problem too.

**Comment by Undin** [comment]
It's fixed in #4914 and will be landed in the next release (I hope).
I close this issue as a duplicate of #4627 because they have the same reason
